src/components/ProjectPage.js

```diff
@@ -8,91 +8,298 @@ import { useNavigate, useParams } from "react-router-dom";
 
 export default function ProjectPage() {
 
-  
-     const authorization = JSON.parse(window.localStorage.getItem('token'));
-     let navigate = useNavigate(); 
-     /*
-     useEffect(() => {
-      if(id) {
-       
-       axiosInstance
-           .get(`/projects/${id}` , {
-               headers: {authorization}, 
-           } 
-               )
-           .then((response) => {
-           })
-           .catch((err) => {
-               console.log(err);
-               console.log(err.response);
-               setErrorMessage(err.response.data.msg); 
-               setError(true);
-           });
-      }
-}, [id]);*/
-  
-    return (
+    const [name, setName] = useState('');
+    const [tags, setTags] = useState([]);
+    const [description, setDescription] = useState('');
+    const [skills, setSkills] = useState([]);
+    const [links, setLinks] = useState([]);
+    const [error, setError] = useState(false);
+    const [canUserEdit, setCanUserEdit] = useState(null);
+    const [canUserFollow, setCanUserFollow] = useState(null);
+    const [canUserRequest, setCanUserRequest] = useState();
+    const [hasUserRequested, setHasUserRequested] = useState();
+    const [isUserFollowing, setIsUserFollowing] = useState();
+    const [isUserAMember, setIsUserAMember] = useState();
+    const [isValidUser, setIsValidUser] = useState();
+    const [members, setMembers] = useState([]);
+    const [status, setStatus] = useState('');
```

Comment:
The project info could have all been stored in state, where the state is an object and each individual state here could have been a key in the object.